com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -286,19 +285,25 @@ public static void NetworkShow(List<NetworkObject> networkObjects, ulong clientI
                     throw new ArgumentNullException("All " + nameof(NetworkObject) + "s must belong to the same " + nameof(NetworkManager));
                 }
             }
+
             using (var buffer = PooledNetworkBuffer.Get())
             using (var writer = PooledNetworkWriter.Get(buffer))
             {
                 writer.WriteUInt16Packed((ushort)networkObjects.Count);
-                long prevLength = 0;
+
+                var initialBufferLength = buffer.Length;
+                var previousBufferLength = initialBufferLength;
+
```

Comment:
Not strictly necessary, but there could be an opportunity for a data structure to encapsulate this pattern. Something like:

```
public struct BufferSizeCapture
{
    private readonly ulong m_InitialLength;
    private readonly Buffer m_Buffer;

    private ulong m_PreviousLength;

    public BufferSizeCapture(Buffer buffer)
    {
        m_Buffer = buffer;
        m_InitialLength= buffer.Length;
        m_PreviousLength= InitialLength;
    }

    public ulong Flush()
    {
        ulong currentLength = buffer.Length;
        ulong retval = currentLength - m_PreviousLength + m_InitialLength;
        m_PreviousLength= currentLength;
        return retval;
    }

}
```

then boilerplate is pretty much

```
var bufferSizeCapture = new BufferSizeCapture(buffer);

...

networkManager.NetworkMetrics.TrackObjectSpawnSent(clientId, networkObjects[i].NetworkObjectId, networkObjects[i].name, bufferSizeCapture.Flush());
```

Reply:
I like this!